lib/srv/termmanager.go

```diff
@@ -62,7 +62,15 @@ func NewTermManager() *TermManager {
 
 func (g *TermManager) writeToClients(p []byte) int {
 	g.history = append(g.history, p...)
-	g.history = g.history[:maxHistory]
+
+	var truncate int
+	if len(g.history) > maxHistory {
+		truncate = maxHistory
+	} else {
+		truncate = len(g.history)
+	}
+
+	g.history = g.history[:truncate]
```

Comment:
We expect it to grow but it shouldn't reach a very large size size it's max size is close to the amount of data that is sent in a single write call. I don't expect it to be very large since we're dealing with terminals and stdout here. What we could do is trash the history buffer and replace it if it blows up by a lot but I don't expect this to be a problem.

Reply:
I don't expect this to become an issue but to handle the pathological case I've added code that truncates p from the front before appending it. This puts a hard limit on `g.history` such that `len(g.history) < 2*maxHistory`.

Reply:
This prevents any giant allocations while not forcing us to periodically allocate new smaller slices if the size exceeds a threshold.